packages/core/src/blockchain/inherent/parachain/validation-data.ts

```diff
@@ -268,6 +268,12 @@ export class SetValidationData implements InherentProvider {
       newEntries.push([upgradeKey, null])
     }
 
+    // Apply relay state overrides
+    newEntries = (params.relayChainStateOverrides || []).reduce(
```

Comment:
this code is not easy to understand. wouldn't it be simpler with for loop?

Reply:
Pushed a more imperative version of the code. 

I usually use reduce where possible to make the code more compact and simpler sometimes.